### PR TITLE
enroot+pyxis

### PR DESCRIPTION
**Pull request by xpillons**
- install enroot on compute nodes
- added scheduler and login cluster-inits
- update template for new specs
- compile pyxis on scheduler
- share spank_pyxis.so on /sched
- group everything in same scripts and call functions based on roles
- enable user namespace for AlmaLinux to fix "no space left on device" enroot error

### CODE REVIEW

File: specs/login/cluster-init/23-pyxis.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (bwatrous) [inline specs/login/cluster-init/23-pyxis.sh:1]: Is this file needed?   It's identical to the script from 'default' (and it's not in the "scripts" directory)

File: specs/scheduler/cluster-init/scripts/23-pyxis.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -e
+PYXIS_VERSION=0.19.0
+SHARED_DIR=/sched/pyxis
+
+function link_plugstack() {
```
Comment (bwatrous) [inline specs/scheduler/cluster-init/scripts/23-pyxis.sh:6]: Could we split this into a new file "22-plugstack.sh" or something similar and then have just one 23-pyxis.sh?
Will it work correctly if the 23-pyxis.sh from default runs before this script?

File: specs/login/cluster-init/23-pyxis.sh (hunk 1 above)
Comment (xpillons) [reply specs/login/cluster-init/23-pyxis.sh:1]: yes the default is used only for execute, I will reorganize the whole project

File: specs/scheduler/cluster-init/scripts/23-pyxis.sh (hunk 2 above)
Comment (xpillons) [reply specs/scheduler/cluster-init/scripts/23-pyxis.sh:6]: yes will try to have a single script per feature which will behave differently based on the instance it runs on

Comment (bwatrous) [reply specs/scheduler/cluster-init/scripts/23-pyxis.sh:6]: Part of the comment was that the file needs to be in the scripts/ directory if it needs to be run :)

File: specs/default/cluster-init/scripts/21-pmix.sh [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
-PMIX_ROOT=/opt/pmix
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "$script_dir/../files/common.sh" 
+read_os
 
-logger -s "Configuring PMIx"
+PMIX_ROOT=/opt/pmix
 
+# TODO: Try not to change the slurmd configuration file, instead use the environment variables in the job script
 function configure_slurmd()
```
Comment (ryanhamel) [inline specs/default/cluster-init/scripts/21-pmix.sh:10]: Can we just remove this now? If not, can we put a # Deprecated comment on it, and explain in what situations someone may need it?

**Comment by xpillons** [comment]
@ryanhamel good to review

**Review by ryanhamel — APPROVED** [review]
